src/test/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/indexstatemanagement/MetadataRegressionIT.kt

```diff
@@ -0,0 +1,241 @@
+            UpdateManagedIndexMetaDataAction.INSTANCE, request
+        ).get()
+
+        logger.info(response.isAcknowledged)
+        indexMetadata = getIndexMetadata(indexName)
+        logger.info("check if metadata is saved in cluster state: ${indexMetadata.getCustomData("managed_index_metadata")}")
+
+        // start the job run
+        updateManagedIndexConfigStartTime(managedIndexConfig)
+        waitFor {
+            assertEquals(
+                "Index did not set number_of_replicas to ${actionConfig.numOfReplicas}",
+                actionConfig.numOfReplicas,
+                getNumberOfReplicasSetting(indexName)
+            )
+        }
+    }
+
+    fun `test new node skip execution when old node exist in cluster`() {
+        Assume.assumeTrue(isMixedNodeRegressionTest)
+
+        /**
+         * mixedCluster-0 is new node, mixedCluster-1 is old node
+         *
+         * set config index to only have one shard on new node
+         * so old node cannot run job because it has no shard
+         * new node also cannot run job because there is an old node
+         * here we check no job can be run
+         *
+         * then reroute shard to old node and this old node can run job
+         */
+
+        val indexName = "${testIndexName}_index_1"
+        val policyID = "${testIndexName}_testPolicyName_1"
+        val actionConfig = ReplicaCountActionConfig(10, 0)
+        val states = listOf(State(name = "ReplicaCountState", actions = listOf(actionConfig), transitions = listOf()))
+        val policy = Policy(
+            id = policyID,
+            description = "$testIndexName description",
+            schemaVersion = 1L,
+            lastUpdatedTime = Instant.now().truncatedTo(ChronoUnit.MILLIS),
+            errorNotification = randomErrorNotification(),
+            defaultState = states[0].name,
+            states = states
+        )
+
+        createPolicy(policy, policyID)
+        createIndex(indexName)
+
+        val settings = Settings.builder()
+            .put(IndexMetadata.SETTING_NUMBER_OF_REPLICAS, "0")
+            .build()
+        updateIndexSettings(INDEX_MANAGEMENT_INDEX, settings)
+
+        // check config index shard position
+        val shardsResponse = catIndexShard(INDEX_MANAGEMENT_INDEX)
+        logger.info("check config index shard: $shardsResponse")
+        val shardNode = (shardsResponse[0] as HashMap<*, *>)["node"]
+
+        sleep(3000) // wait some time for cluster to be stable
+
+        // move shard on node1 to node0 if exist
+        if (shardNode == "mixedCluster-1") rerouteShard(INDEX_MANAGEMENT_INDEX, "mixedCluster-1", "mixedCluster-0")
+
+        addPolicyToIndex(indexName, policyID)
+
+        val managedIndexConfig = getExistingManagedIndexConfig(indexName)
+        updateManagedIndexConfigStartTime(managedIndexConfig)
+
+        // check no job has been run
+        wait { assertEquals(null, getExistingManagedIndexConfig(indexName).policy) }
```

Comment:
So best case (in successful path) this just runs the assert block every 100ms for 10 seconds to verify nothing ran? Why not just sleep the thread for 10 seconds and assert it never ran instead of a new function etc.?